### PR TITLE
refactor(components): migrate Head component from next/head to next-seo

### LINKED ISSUE #3838: Replace <Head> with garmeeh/next-seo

### PR DESCRIPTION
**Pull request by byhow**
referencing github issue #3838 and recommendation from next-seo docs that next-seo is better suited for this sort of things over next/head. most of the things here overlaps with app router's [metadata API](https://nextjs.org/docs/app/building-your-application/optimizing/metadata) too.

fix #3838

### CODE REVIEW

File: src/components/Head/index.tsx [hunk 1]
```diff
@@ -32,204 +34,206 @@ interface HeadProps {
...
+          : `matters,${process.env.NEXT_PUBLIC_SITE_DOMAIN},創作有價`,
+      },
+      {
+        name: 'viewport',
+        content:
+          'minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no, user-scalable=no, viewport-fit=cover',
+      },
+      {
+        name: 'monetization',
+        content: props.paymentPointer || '',
```
Comment (robertu7) [inline src/components/Head/index.tsx:136]: Will `next-seo` render meta tags that `content` is nullish?

**Review by robertu7 — APPROVED** [review]
LGTM! 

Changed the based branch and will open another PR for Vercel preview.

File: src/components/Head/index.tsx (hunk 1 above)
Comment (robertu7) [reply src/components/Head/index.tsx:136]: can confirm empty tags are still be rendered.

image

Comment (byhow) [reply src/components/Head/index.tsx:136]: Ah good catch! yeah this static config way of things isn't the best for ternary operations

Comment (byhow) [reply src/components/Head/index.tsx:136]: temporary resort can be that we keep the ternary stuff still in `next/head` if we don't want to conditionally merging jsons all the time

Comment (robertu7) [reply src/components/Head/index.tsx:136]: Can we use object spread with ternary?

```js
...props.paymentPointer ? { name: 'monetization', content: props.paymentPointer } : {},
...isProdServingCanonical ? {} : { name: 'googlebot', content: 'noindex, nofollow'},
```

**Pull request merged**